src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private double radius;
+
+    public Circle(String color, double radius) {
+        super(color);
+        this.radius = radius;
+    }
+
+    @Override
+    public double findArea() {
+        if (getArea() == 0.0) {
+            return radius * radius * Math.PI;
+        } else {
+            System.out.println("The area has been already calculated");
+            return getArea();
+        }
+    }
```

Comment:
now `getArea` will always return 0
we can
- just return `radius * radius * Math.PI` and remove unnecessary code and `area` field from `Figure` class
- or set area if it wasn't calculated and get it, if it was 
(for this we should add `getArea` and `setArea` to `Figure` class, but with `protected` modifier and use these methods in `findArea`)
(same for other classes)